sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py

```diff
@@ -0,0 +1,82 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from azure.core.exceptions import HttpResponseError
+from azure.core.polling.base_polling import (
+    LocationPolling,
+    OperationResourcePolling,
+    _is_empty,
+    _as_json,
+    BadResponse
+)
+
+
+class TrainingPolling(LocationPolling):
+    """Polling method overrides for training endpoints.
+    """
+
+    def get_polling_url(self):
+        # type: () -> str
+        """Return the polling URL.
+        """
+        return self._location_url + "?includeKeys=true"
+
+    def get_status(self, pipeline_response):  # pylint: disable=no-self-use
+        # type: (PipelineResponseType) -> str
+        """Process the latest status update retrieved from a 'location' header.
+
+        :param azure.core.pipeline.PipelineResponse response: latest REST call response.
+        :raises: BadResponse if response has no body.
+        """
+        response = pipeline_response.http_response
+        if response.status_code == 200:
+            body = _as_json(response)
+            status = body['modelInfo']['status']
+            if not status:
+                raise BadResponse("No status found in body")
```

Comment:
Was wondering given that the service returns `200 OK` on bad input for analyze on forms endpoint. Should we directly be returning the corresponding [AnalyzeOperationResult.status](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json#L516) for polling status rather than checking the http status code here?

Reply:
The code you highlighted is for TrainingPolling. I do poll on AnalyzeOperationResult.status in the [AnalyzePolling](https://github.com/Azure/azure-sdk-for-python/pull/10420/files/08069c856ad97d3d49c4653bd0272c15cff82325#diff-483c5b2f3560dbcc72dde39455d0668eR70) method.

For training polling, the service returns 200 even if the model is still creating or invalid. So I use the ModelInfo.status to poll.